sdk/src/test/java/com/hedera/hashgraph/sdk/ClientTest.java

```diff
@@ -134,21 +138,101 @@ void testFromJson() throws Exception {
+        Assertions.assertEquals(network.get("1.testnet.hedera.com:50211"), new AccountId(4));
 
-        client.setNetwork(setNetworkNodes);
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(2, network.size());
+        Assertions.assertEquals(network.get("0.testnet.hedera.com:50211"), new AccountId(3));
+        Assertions.assertEquals(network.get("1.testnet.hedera.com:50211"), new AccountId(4));
+
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(5));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(3, network.size());
+        Assertions.assertEquals(network.get("0.testnet.hedera.com:50211"), new AccountId(3));
+        Assertions.assertEquals(network.get("1.testnet.hedera.com:50211"), new AccountId(4));
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(5));
+
+        nodes.clear();
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(5));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(1, network.size());
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(5));
+
+        nodes.clear();
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(6));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(1, network.size());
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(6));
+
+        client.close();
+    }
+
+    @Test
+    @DisplayName("setNetwork() functions correctly")
+    void setMirrorNetworkWorks() throws Exception {
+        var nodes = new ArrayList<String>();
+        nodes.add("hcs.testnet.mirrornode.hedera.com:5600");
+
+        Client client = Client.forNetwork(new HashMap<>()).setMirrorNetwork(nodes);
+        @Var var mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(1, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+
+        client.setMirrorNetwork(nodes);
+        mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(1, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+
+        nodes.add("hcs.testnet1.mirrornode.hedera.com:5600");
+
+        client.setMirrorNetwork(nodes);
+        mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(2, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet1.mirrornode.hedera.com:5600"));
+
+        nodes.clear();
+        nodes.add("hcs.testnet1.mirrornode.hedera.com:5600");
```

Comment:
Shouldn't reuse variables. It makes it hard to follow the tests. Better to create a new variable with a descriptive name.

```suggestion
        final var deletedNode = List.of("hcs.testnet1.mirrornode.hedera.com:5600");
```

Reply:
I will update this, but just wanted to know the rational behind using a single variable. Using a single variable which is manipulated makes sure `setMirroNetwork()` works correctly even when the same list pointer is passed into it several times. Perhaps the entire test shouldn't need to do this, but I feel we should have at least one test to confirms an updated list will change the current network.

Reply:
Sure, you can keep one test doing that. Probably the test that verifies it is unchanged when the same network is passed into it again would make the most sense.